modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/ResourceTypeResourceManager.java

```diff
@@ -15,213 +15,188 @@
+                rt -> rt.getId().equals(id)).findAny().orElse(null);
+            if (resourceType == null) {
+                throw new NotFoundException("resource with id '" + id + "' does not exist");
             }
-            //put the uri of the resource type object in the response header parameter.
-            return new SCIMResponse(ResponseCodeConstants.CODE_OK,
-                    encodedObject, responseHeaders);
-        } catch (CharonException e) {
-            return encodeSCIMException(e);
-        } catch (BadRequestException e) {
-            return encodeSCIMException(e);
-        } catch (InternalErrorException e) {
-            return encodeSCIMException(e);
-        } catch (NotFoundException e) {
-            return encodeSCIMException(e);
-        } catch (JSONException e) {
-            return null;
+            rethrowConsumer(rt -> validateResourceTypeSCIMObject((AbstractSCIMObject) rt)).accept(resourceType);
+            String encodedObject = getEncoder().encodeSCIMObject(resourceType);
+            Map<String, String> responseHeaders = new HashMap<>();
+            responseHeaders.put(SCIMConstants.LOCATION_HEADER,
+                getResourceEndpointURL(SCIMConstants.RESOURCE_TYPE_ENDPOINT));
+            responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+            return new SCIMResponse(ResponseCodeConstants.CODE_OK, encodedObject, responseHeaders);
+        } catch (AbstractCharonException ex) {
+            return encodeSCIMException(ex);
+        } catch (RuntimeException ex) {
+            log.error(ex.getMessage(), ex);
+            CharonException charonException = new CharonException("an unexpected error occured: " + ex.getMessage());
+            return encodeSCIMException(charonException);
         }
     }
 
     @Override
-    public SCIMResponse create(String scimObjectString, UserManager userManager, String attributes, String
-            excludeAttributes) {
+    public SCIMResponse create(String scimObjectString, String attributes, String excludeAttributes) {
 
         String error = "Request is undefined";
         BadRequestException badRequestException = new BadRequestException(error, ResponseCodeConstants.INVALID_PATH);
         return encodeSCIMException(badRequestException);
     }
 
     @Override
-    public SCIMResponse delete(String id, UserManager userManager) {
+    public SCIMResponse delete(String id) {
 
         String error = "Request is undefined";
         BadRequestException badRequestException = new BadRequestException(error, ResponseCodeConstants.INVALID_PATH);
         return encodeSCIMException(badRequestException);
     }
 
     @Override
-    public SCIMResponse listWithGET(UserManager userManager, String filter, int startIndex, int count, String sortBy,
+    public SCIMResponse listWithGET(String filter, Integer startIndexInt, Integer countInt, String sortBy,
```

Comment:
Changing public APIs are not recommended.  There may be users who use this. Changes should be done in a backward-compatible manner.